tests/e2e/support/objects/app-files/resource/actions.ts

```diff
@@ -91,66 +155,21 @@ export const createResource = async (args: createResourceArgs): Promise<void> =>
 
     if (!resourcesExists) {
       await page.locator(addNewResourceButton).click()
-
-      switch (type) {
-        case 'folder': {
-          await page.locator(createNewFolderButton).click()
-          await page.locator(resourceNameInput).fill(resource)
-          await Promise.all([
-            page.waitForResponse(
-              (resp) => resp.status() === 207 && resp.request().method() === 'PROPFIND'
-            ),
-            page.locator(actionConfirmationButton).click()
-          ])
-          break
-        }
-        case 'txtFile': {
-          await page.locator(createNewTxtFileButton).click()
-          await page.locator(resourceNameInput).fill(resource)
-          await Promise.all([
-            page.waitForResponse(
-              (resp) => resp.status() === 201 && resp.request().method() === 'PUT'
-            ),
-            page.locator(actionConfirmationButton).click()
-          ])
-          await editTextDocument({ page, content: content })
-          break
-        }
-        case 'mdFile': {
-          await page.locator(createNewMdFileButton).click()
-          await page.locator(resourceNameInput).fill(resource)
-          await Promise.all([
-            page.waitForResponse(
-              (resp) => resp.status() === 201 && resp.request().method() === 'PUT'
-            ),
-            page.locator(actionConfirmationButton).click()
-          ])
-          await editTextDocument({ page, content: content })
-          break
-        }
-        case 'drawioFile': {
-          await page.locator(createNewDrawioFileButton).click()
-          await page.locator(resourceNameInput).fill(resource)
-
-          const [drawioTab] = await Promise.all([
-            page.waitForEvent('popup'),
-            page.waitForResponse(
-              (resp) => resp.status() === 201 && resp.request().method() === 'PUT'
-            ),
-            page.locator(actionConfirmationButton).click()
-          ])
-          await drawioTab.waitForLoadState()
-          await drawioTab.locator('.geBigButton', { hasText: 'Save' }).isVisible()
-          await drawioTab.waitForURL('**/draw-io/personal/**')
-          await drawioTab.close()
-          break
-        }
-      }
-    }
-    if (type === 'folder') {
+      await createNewFolder({ page, resource: resource })
+      await clickResource({ page, path: resource })
+    } else {
       await clickResource({ page, path: resource })
     }
   }
+
+  const baseResourcesExists = await resourceExists({
+    page: page,
+    name: baseResource
+  })
+
+  if (!baseResourcesExists) {
+    await createNewFileOrFolder({ page, name: baseResource, type: type, content: content })
+  }
```

Comment:
`baseResource` shouldn't exist and the goal of the function is to create that resource so I don't think we need to check its existence.
```suggestion
  await createNewFileOrFolder({ page, name: baseResource, type: type, content: content })
```

Reply:
@saw-jan I ask with **Viktor** .He told me it will be better if we can create resources in the same step if baseResource doesn't exist. So I modified the create resource to step in such a way that if the base resource does not exist create the resource and navigate to that resource.

Reply:
what if base resources already exist before creation @saw-jan bro?

Reply:
> what if base resources already exist before creation @saw-jan bro?

We don't need to assert those conditions. Our sole purpose should be only to create the resource. If it already exists then the When step will fail.

Reply:
`throw new Error(`${baseResource} already exists`)` i think we can use this. if not why? @saw-jan

Reply:
My main point is we don't event need to check its existence.
So only `await createNewFileOrFolder({ page, name: baseResource, type: type, content: content })` should be enough

Reply:
my point is that throwing some exact error message is good than just a random selector not visible message.
I don't have a complete idea of how e2e tests are structured so I am also confused here. @individual-it  and @SwikritiT could you help us here?

Reply:
I agree with @saw-jan, I don't think we need to check `baseResource` if it's the one we're intending to create. With `Parent` resource it makes sense as we might want to create multiple `child-resources`.Like sajan said the when step should fail in the case where a resource already exists. And we explicitly provide in the feature file the resources we want to create, so IMO feature should be written in such a way that it shouldn't happen unless it's intended(like testing to create a resource with the same name, will this code throw an error in this case?).